dvc/remote/ssh/connection.py

```diff
@@ -282,3 +288,17 @@ def open_max_sftp_channels(self):
                         raise
                     break
         return self._sftp_channels
+
+    def open(self, *args, **kwargs):
+        return self.sftp.open(*args, **kwargs)
+
+    def symlink(self, *args, **kwargs):
+        return self.sftp.symlink(*args, **kwargs)
```

Comment:
does sftp.symlink create dirs automatically? I see that you are using self.makedirs in reflink and hardlink, but not here.

Reply:
good observation, I don't think it does, I'll use `makedirs` too.

Reply:
@mroutis but isn't link() already running makedirs?

Reply:
yes, @efiop , this is something that we should decide on.
If we are going to `makedirs` on `copy`/`hardlink`/`reflink`/`symlink` or we are going to trust the programmer to make sure the destination already has their parent directories created.

This trust thingy is something that happens across the code base, for example, some methods have assertions like the `isfile`, `exists`, and things like that.

It would be interesting to hear other's opinions on this aspect ^ (cc: @Suor , @pared , @shcheklein)

Reply:
@mroutis we are already creating dirs automatically on creating a link in link() method where it is justified, I'm just saying that these low-level methods shouldn't do that.